src/ORM/FieldType/DBClassName.php

```diff
@@ -6,48 +6,40 @@
 use SilverStripe\Core\Config\Config;
 use SilverStripe\ORM\DataObject;
 use SilverStripe\ORM\DB;
+use SilverStripe\View\ViewableData;
 
 /**
  * Represents a classname selector, which respects obsolete clasess.
  */
 class DBClassName extends DBEnum
 {
-
     /**
      * Base classname of class to enumerate.
      * If 'DataObject' then all classes are included.
      * If empty, then the baseClass of the parent object will be used
-     *
-     * @var string|null
      */
-    protected $baseClass = null;
+    protected ?string $baseClass = null;
 
     /**
      * Parent object
-     *
-     * @var DataObject|null
      */
-    protected $record = null;
+    protected ?DataObject $record = null;
 
-    private static $index = true;
+    private static string|bool $index = true;
```

Comment:
See the PHPDoc for this property in `DBField` for the reason behind this typing.